pypdf/generic/_data_structures.py

```diff
@@ -121,6 +128,43 @@ def items(self) -> Iterable[Any]:
         """Emulate DictionaryObject.items for a list (index, object)."""
         return enumerate(self)
 
+    def __to_lst(self, lst: Any) -> List[Any]:
+        # Convert to list, internal
+        if isinstance(lst, (list, tuple, set)):
+            pass
+        elif isinstance(lst, PdfObject):
+            lst = [lst]
+        elif isinstance(lst, str):
+            if lst[0] == "/":
+                lst = [NameObject(lst)]
+            else:
+                lst = [TextStringObject(lst)]
+        elif isinstance(lst, bytes):
+            lst = [ByteStringObject(lst)]
+        else:  # for numbers,...
+            lst = [lst]
+        return lst
+
+    def __add__(self, lst: Any) -> "ArrayObject":
+        """Allow extend with any list format or append"""
+        return ArrayObject(self) + ArrayObject(self.__to_lst(lst))
+
+    def __iadd__(self, lst: Any) -> Self:
+        """Allow extend with any list format or append"""
+        for x in self.__to_lst(lst):
+            self.append(x)
```

Comment:
What about using `self.extend(self.__to_lst(lst)` directly?

Reply:
was not sure about types mixup. Will follow your advice